biofefi/pages/2_Data_Preprocessing.py

```diff
@@ -0,0 +1,325 @@
+
+    if feature_selection_methods[ConfigStateKeys.VarianceThreshold]:
+        varianceselector = VarianceThreshold(
+            threshold=st.session_state[ConfigStateKeys.ThresholdVariance]
+        )
+        X = varianceselector.fit_transform(X)
+        variance_columns = varianceselector.get_feature_names_out()
+        X = pd.DataFrame(X, columns=variance_columns)
+
+    if feature_selection_methods[ConfigStateKeys.CorrelationThreshold]:
+        corr_matrix = X.corr().abs()
+        upper_triangle = corr_matrix.where(
+            np.triu(np.ones(corr_matrix.shape), k=1).astype(bool)
+        )
+        to_drop = [
+            column
+            for column in upper_triangle.columns
+            if any(
+                upper_triangle[column]
+                > st.session_state[ConfigStateKeys.ThresholdCorrelation]
+            )
+        ]
+        X = X.drop(columns=to_drop)
+
+    if feature_selection_methods[ConfigStateKeys.LassoFeatureSelection]:
+        lasso = Lasso(alpha=st.session_state[ConfigStateKeys.RegularisationTerm])
+        lasso.fit(X, y)
+        selected_features = X.columns[lasso.coef_ != 0]
+        X = X[selected_features]
+
+    data = pd.concat([X, y], axis=1)
+
+
+def normalise_independent_variables(normalisation_method: str, X):
+
+    if normalisation_method == Normalisations.NoNormalisation:
+        return X
+
+    elif normalisation_method == Normalisations.Standardization:
+        scaler = StandardScaler()
+
+    elif normalisation_method == Normalisations.MinMax:
+        scaler = MinMaxScaler()
+
+    column_names = X.columns
+    X = scaler.fit_transform(X)
+
+    X = pd.DataFrame(X, columns=column_names)
+
+    return X
+
+
+def transform_dependent_variable(transformation_y_method: str, y):
+
+    if transformation_y_method == TransformationsY.NoTransformation:
+        return y
+
+    column_name = y.name
+    y = y.to_numpy().reshape(-1, 1)
+
+    if transformation_y_method == TransformationsY.Log:
+        if y.min() <= 0:
+            y = y - y.min() + 1
+        y = np.log(y)
+
+    elif transformation_y_method == TransformationsY.Sqrt:
+        if y.min() < 0:
+            y = y - y.min()
+        y = np.sqrt(y)
+
+    elif transformation_y_method == TransformationsY.MinMaxNormalisation:
+        scaler = MinMaxScaler()
+        y = scaler.fit_transform(y)
+
+    elif transformation_y_method == TransformationsY.StandardisationNormalisation:
+        scaler = StandardScaler()
+        y = scaler.fit_transform(y)
+
+    y = pd.DataFrame(y, columns=[column_name])
+
+    return y
+
+
+def run_preprocessing(data: pd.DataFrame, experiment_path: Path) -> None:
+
+    X = data.iloc[:, :-1]
+    y = data.iloc[:, -1]
+
+    config = build_config()
+    save_options(data_preprocessing_options_path(experiment_path), config)
+    X = normalise_independent_variables(config.dependent_variable_normalisation, X)
+    y = transform_dependent_variable(config.independent_variable_transformation, y)
```

Comment:
should these be the other way around? `normalise_independent_variables(config.independent_variable_normalisation, X)` and `transform_dependent_variable(config.dependent_variable_transformation, y)`